model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/CacheManager.java

```diff
@@ -178,8 +229,7 @@ public void addRevisioned(Revisioned object, long startupRevision, long lifespan
             }
             // revisions cache has a lower value than the object.revision, so update revision and add it to cache
             revisions.put(id, object.getRevision());
-            if (lifespan < 0) cache.putForExternalRead(id, object);
-            else cache.putForExternalRead(id, object, lifespan, TimeUnit.MILLISECONDS);
+            cache.putForExternalRead(id, object, lifespan, TimeUnit.MILLISECONDS, idleTime, TimeUnit.MILLISECONDS);
```

Comment:
I think we can drop the if branch here since any value < 0 will disable the feature still in infinispan. 

Also added idleTime here